Sources/AblyChat/DefaultPresence.swift

```diff
@@ -0,0 +1,234 @@
+import Ably
+
+@MainActor
+internal final class DefaultPresence: Presence, EmitsDiscontinuities {
+    private let featureChannel: FeatureChannel
+    private let roomID: String
+    private let clientID: String
+    private let logger: InternalLogger
+
+    internal init(featureChannel: FeatureChannel, roomID: String, clientID: String, logger: InternalLogger) {
+        self.roomID = roomID
+        self.featureChannel = featureChannel
+        self.clientID = clientID
+        self.logger = logger
+    }
+
+    internal nonisolated var channel: any RealtimeChannelProtocol {
+        featureChannel.channel
+    }
+
+    // (CHA-PR6) It must be possible to retrieve all the @Members of the presence set. The behaviour depends on the current room status, as presence operations in a Realtime Client cause implicit attaches.
+    internal func get() async throws -> [PresenceMember] {
+        logger.log(message: "Getting presence", level: .debug)
+        return try await withCheckedThrowingContinuation { continuation in
+            channel.presence.get { [processPresenceGet] members, error in
+                do {
+                    try processPresenceGet(continuation, members, error)
```

Comment:
I think it's a bit extra to pack the continuation into this func, it's easy to call it here.

Reply:
fixed